node/src/accountant/scanners_tools.rs

```diff
@@ -369,8 +381,10 @@ mod tests {
     fn payable_above_threshold_values_is_marked_qualified_and_returns_threshold() {
         let now = SystemTime::now();
         let payment_thresholds = PaymentThresholds::default();
-        let debt = payment_thresholds.permanent_debt_allowed_gwei + 1_000_000_000;
-        let time = to_time_t(now) - payment_thresholds.maturity_threshold_sec - 1;
+        let debt = payment_thresholds.debt_threshold_gwei - 1;
+        let time = payment_thresholds.maturity_threshold_sec
+            + payment_thresholds.threshold_interval_sec
+            - 1;
```

Comment:
These values are way safe! If I were you I'd check if there are still tests that test the effect of being above or under the declining part of the payment threshold somehow more seriously.

Reply:
I see. Still, I don't know which values will get to the corner case.

Reply:
Yeah... I have serious concerns here. I will get into them...
As for your test it struck my mind that instead of the variable name `threshold` and welcome `point_on_the_slope`.

However, now the important part. I have reasons to think that during your merge with master some test were lost. These tests in fact used to provide the tight assertion which I've been demanding from you. Do you see my previous comment about the tested value picked too far from the threshold? 

I will mention them (I afraid I could miss some :frowning: potentially). These are either tests or functions which are in master but not in your code and I don't see a reason why they shouldn't be considered belonging with there anymore (I actually did really big effort to document the thresholds with tests because we had lacked any good idea if it worked wrong or right): 

`payables_debug_summary_stays_still_if_no_qualified_payments`,
`payout_sloped_segment_in_payment_thresholds_goes_along_proper_line`,
`gap_tester`,
`assert_on_height_granularity_with_advancing_time`,
`testing_granularity_calculate_sloped_threshold_by_time`,
`checking_chosen_values_for_the_payment_thresholds_defaults_on_height_values_granularity`,
`slope_has_loose_enough_limitations_to_allow_work_with_number_bigger_than_masq_token_max_supply`,
`slope_after_its_end_turns_into_permanent_debt_allowed`,
`is_innocent_age_works_for_age_smaller_than_innocent_age`,
`is_innocent_age_works_for_age_equal_to_innocent_age`,
`is_innocent_age_works_for_excessive_age`
`is_innocent_balance_works_for_balance_smaller_than_innocent_balance`,
`is_innocent_balance_works_for_balance_equal_to_innocent_balance`,
`is_innocent_balance_works_for_excessive_balance`,
`payable_is_found_innocent_by_age_and_returns`,
`payable_is_found_innocent_by_balance_and_returns`,
`threshold_calculation_depends_on_user_defined_payment_thresholds`

Aside those above, which are all a must, I'd like to see you use the master version of :
`payables_debug_summary` (ignoring the fact I prefer functional programming to imperative one, the structure of the function answers the extra tools I created for testing the threshold mechanism and so it requires `self` of Accountant and so on...). I'd really appreciate if you simply pick it up from master and put it into your branch. 
Together with this function there are also two tests which don't have an alternative in your code. They are:
`payables_debug_summary_prints_pretty_summary `,
`payables_debug_summary_stays_still_if_no_qualified_payments` 

I also wish you to answer this comment in the sense of "done" and then let it open so that I can compare it with the changes submitted for the next iteration. 

Maybe I will have to have a look at the commits where you were doing the merge of master. Just to be sure there aren't more places where you overlooked that there was new piece of code added.

Reply:
1) `payables_debug_summary_stays_still_if_no_qualified_payments` -> `qualified_payables_can_be_filtered_out_from_non_pending_payables_along_with_their_summary`

2) `payout_sloped_segment_in_payment_thresholds_goes_along_proper_line` -> `New Code`

3) `gap_tester` -> `New Code`

4) `assert_on_height_granularity_with_advancing_time` -> `New Code`

5) `testing_granularity_calculate_sloped_threshold_by_time` -> `New Code`

6) `checking_chosen_values_for_the_payment_thresholds_defaults_on_height_values_granularity` -> `New Code`

7) `slope_has_loose_enough_limitations_to_allow_work_with_number_bigger_than_masq_token_max_supply` -> `New Code`

8) `slope_after_its_end_turns_into_permanent_debt_allowed` -> `New Code`

9) `is_innocent_age_works_for_age_smaller_than_innocent_age` -> `New Function - is_payable_qualified()`

10) `is_innocent_age_works_for_age_equal_to_innocent_age` -> `New Function - is_payable_qualified()`

11) `is_innocent_age_works_for_excessive_age` -> `New Function - is_payable_qualified()`

12) `is_innocent_balance_works_for_balance_smaller_than_innocent_balance` -> `New Function - is_payable_qualified()`

13) `is_innocent_balance_works_for_balance_equal_to_innocent_balance` -> `New Function - is_payable_qualified()`

14) `is_innocent_balance_works_for_excessive_balance` -> `New Function - is_payable_qualified()`

15) `payable_is_found_innocent_by_age_and_returns` -> `New Function - is_payable_qualified()`

16) `payable_is_found_innocent_by_balance_and_returns` -> `New Function - is_payable_qualified()`

17) `threshold_calculation_depends_on_user_defined_payment_thresholds` -> `Removed due to the change in the way things are used`

Reply:
You're right on that :point_up:. I just wanna add that our preferred state is to keep the next tests as simple as possible and therefore don't use any higher-level function in the tests but only these bare functions whose names stand in the names of the tests: 

    is_innocent_age_works_for_age_smaller_than_innocent_age

    is_innocent_age_works_for_age_equal_to_innocent_age

    is_innocent_age_works_for_excessive_age

    is_innocent_balance_works_for_balance_smaller_than_innocent_balance

    is_innocent_balance_works_for_balance_equal_to_innocent_balance

    is_innocent_balance_works_for_excessive_balance

Reply:
Ok!